lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -75,43 +92,114 @@ def mock_unpack(self, dataset_id):
         monkeypatch.setattr(
             "pbench.server.unpack_tarballs.get_tarball_md5", lambda tb: "my_md5_string"
         )
         monkeypatch.setattr(FileTree, "unpack", mock_unpack)
-        obj = Unpack(server_config, self.linksrc, make_logger)
-        total, success = obj.unpack_tarball()
-        assert (
-            f"Unpacking of tarball {self.tar_name} failed: An error occurred while unpacking tarball: tar exited with status 1"
-            in caplog.text
-        )
-        assert total == 1
-        assert success == 0
+        obj = Unpack(server_config, make_logger)
+        result = obj.unpack()
+        for tar in self.tarlist:
+            assert (
+                f"Unpacking of tarball {Path(tar).name} failed: An error occurred while unpacking tarball: tar exited with status 1"
+                in caplog.text
+            )
+        assert result.ntotal == 2
+        assert result.nsuccess == 0
+
+    def test_unpack_symlinks_creation(
+        self, monkeypatch, server_config, make_logger, caplog
+    ):
+        """Show that when symlink creation failes and raises
+        FileNotFoundError Exception it is handled successfully."""
 
-    def test_unpack_tarballs_success(
+        def mock_symlink(path, dest):
+            """
+            Mock of os.symlink() module
+
+            Args:
+                path: tarball file path
+                dest: destination path to create symlink
+
+            Returns:
+                Raises FileNotFoundError Exception
+            """
+
+            raise FileNotFoundError(f"'{path}' -> '{dest}'")
+
+        linkedlistdest = server_config.get("pbench-server", "unpacked-states").split(
+            ","
+        )
+        monkeypatch.setattr(glob, "iglob", lambda x: self.tarlist)
+        monkeypatch.setattr(Path, "resolve", lambda self, strict: self)
+        monkeypatch.setattr(
+            "pbench.server.unpack_tarballs.get_tarball_md5", lambda tb: "my_md5_string"
+        )
+        monkeypatch.setattr(FileTree, "unpack", lambda self, dataset_id: None)
+        monkeypatch.setattr(os, "symlink", mock_symlink)
+        obj = Unpack(server_config, make_logger)
+        result = obj.unpack()
+        for dest in linkedlistdest:
+            for tar in self.tarlist:
+                assert (
+                    f"Error in creation of symlink. ''{server_config.ARCHIVE / 'ABC' / Path(tar).name}' -> '{server_config.ARCHIVE / 'ABC' / dest / Path(tar).name}''"
+                    in caplog.text
+                )
+        assert result.ntotal == 2
+        assert result.nsuccess == 0
+
+    def test_unpack_symlinks_removal(
         self, monkeypatch, server_config, make_logger, caplog
     ):
-        """Show that when unpacking goes smoothly we can count the
-        successfully unpacked tarballs."""
+        """Show that when unlinking of tarball does not goes smoothly
+        and raises FileNotFoundError, it is handled successfully."""
```

Comment:
s/goes/go/

Or, better yet, just go with, "Show that, when unlinking the tarball results in a FileNotFoundError, it is handled successfully."